monday/query_joins.py

```diff
@@ -401,6 +401,46 @@ def get_columns_by_board_query(board_ids):
         }''' % board_ids
 
 
+def duplicate_board_query(
+    board_id: int,
+    duplicate_type: DuplicateTypes,
+    board_name: str = None,
+    workspace_id: int = None,
+    folder_id: int = None,
+    keep_subscribers: bool = None,
+) -> str:
+    board_name = board_name if board_name else ""
+    workspace_id = workspace_id if workspace_id else None
+    folder_id = folder_id if folder_id else None
+    keep_subscribers = keep_subscribers if keep_subscribers else False
+
+    params = """board_id: %s, duplicate_type: %s, board_name: \"%s\"""" % (
+        board_id,
+        duplicate_type.value,
+        board_name,
+    )
+
+    if workspace_id:
+        params += """,  workspace_id: %s"""
+
+    query = """
```

Comment:
This looks like it should be indented once more.

Reply:
Do you mean to move it under the `workspace_id` condition? That condition is meant to only add to the params variable the `workspace_id` parameter for the query but it's not a mandatory condition for the query.

However, there is a clear issue there because there isn't a value passed for the `%s` token, and the test didn't catch it because it was expecting to find a 1 and a 2 within the query, but both are included in the `board_id` value (Its value is 12) That was a really bad test.

I have updated both the query and its test to fix these issues. I wonder if it had something to do with the merge when the branch was like +20 commits behind...

Reply:
Honestly, now that I look at it again I'm not sure what I was talking about, it looks fine... disregard.

Good catch on test, and ty for fixing.